org.testeditor.web.backend.persistence/src/main/java/org/testeditor/web/backend/persistence/exception/PersistenceExceptionMapper.xtend

```diff
@@ -1,18 +1,35 @@
 package org.testeditor.web.backend.persistence.exception
 
 import io.dropwizard.jersey.errors.LoggingExceptionMapper
+import javax.ws.rs.core.Context
 import javax.ws.rs.core.Response
+import javax.ws.rs.core.UriInfo
 import javax.ws.rs.ext.Provider
+import org.testeditor.web.backend.persistence.ConflictingModificationsException
+import org.testeditor.web.backend.persistence.DocumentProvider
+import java.net.URI
 
 @Provider
 class PersistenceExceptionMapper extends LoggingExceptionMapper<PersistenceException> {
 
+	@Context UriInfo uriInfo
+
 	def dispatch toResponse(MaliciousPathException e) {
 		val logId = logException(e)
 		val message = String.format("You are not allowed to access this resource. Your attempt has been logged (ID %016x).", logId);
 		return Response.status(Response.Status.FORBIDDEN).entity(message).build
 	}
 
+	def dispatch Response toResponse(ConflictingModificationsException it) {
+		logException
+		val response = Response.status(Response.Status.CONFLICT).entity(message)
```

Comment:
I'm not really sure whether 'conflict' is the right status here. I interpret it as a status code that indicates that the whole request was aborted and no change took place. This is not exactly what we do, right? The status code OK would not be appropriate either. We might consider a combination of the two embedded in a multi status code (see https://httpstatuses.com/207)

Reply:
I just considered the refactoring we talked about: leave the backend in a conflict state and report it (with HTTP status code CONFLICT, which would then be more appropriate), to give the client the opportunity to decide how to handle it, and offering an endpoint for conflict resolution.

I still consider this a good way to go about it, but I think it deserves its own separate issue. I'd suggest to first complete the current feature as-is, and meet our sprint's objectives, and then refactor (either in this, or the next sprint).